src/logic/wallets/ethTransactions.ts

```diff
@@ -41,7 +42,8 @@ export const calculateGasPrice = async (): Promise<string> => {
 
 export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const ethAdapter = await getSDKWeb3Adapter()
+    // Estimation is a read-only method and we therefore do not need actual address
+    const ethAdapter = getSDKWeb3Adapter(ZERO_ADDRESS)
```

Comment:
Like discussed in the other thread, I don't see the need to pass the address implicitly. Wouldn't the currently connected address work in this case?

Reply:
Depends on how you retrieve it. The initial problem was that we used `getAccountFrom` function that could return null, this is why it falls back to ZERO_ADDRESS. Because this particular method is read-only, we don't care about the address, so I suggested passing the ZERO one.

The suggestion to take the address from redux is not particularly helpful for cases when the function is called outside React component where redux is not easily accessible.

Reply:
That's a good point.

Reply:
In this case, `calculateGasOf` received `txConfig.from` which is `signerAddress`. Accessing the store directly isn't good practice, so I am also tending towards sending the `signerAddress` as an argument. What do you think of my [latest commit](https://github.com/gnosis/safe-react/pull/2745/commits/d11df3307b784863f2c4b559b572c5668ad5c75b)?

Reply:
I think it's fine for now, let's see how it goes in subsequent SDK integrations, maybe some pattern will emerge. We'll probably need the current account address more often than not.